app/scripts/lib/transaction/hooks/enforce-simulation/utils/delegation.ts

```diff
@@ -90,18 +130,92 @@ export function generateDelegation({
 
   for (const tokenBalanceChange of simulationData.tokenBalanceChanges) {
     const tokenAddress = tokenBalanceChange.address;
+    const { standard } = tokenBalanceChange;
     const delta = BigInt(tokenBalanceChange.difference);
-    if (tokenBalanceChange.isDecrease) {
-      caveats = caveats.addCaveat(
-        'erc20MaxLoss',
-        accountAddress,
-        tokenAddress,
-        delta,
-      );
-      log('Caveat - ERC-20 Max Loss', { accountAddress, tokenAddress, delta });
-    } else {
-      caveats = caveats.addCaveat('erc20BalanceGte', tokenAddress, delta);
-      log('Caveat - ERC-20 Balance GTE', { tokenAddress, delta });
+    const tokenId = tokenBalanceChange.id;
+    switch (standard) {
+      case SimulationTokenStandard.erc20:
+        if (tokenBalanceChange.isDecrease) {
+          caveats = caveats.addCaveat(
+            'erc20MaxLoss',
+            accountAddress,
+            tokenAddress,
+            delta,
+          );
+          log('Caveat - ERC-20 Max Loss', {
+            accountAddress,
+            tokenAddress,
+            delta,
+          });
+        } else {
+          caveats = caveats.addCaveat('erc20BalanceGte', tokenAddress, delta);
+          log('Caveat - ERC-20 Balance GTE', { tokenAddress, delta });
+        }
+        break;
+      case SimulationTokenStandard.erc721:
+        if (tokenBalanceChange.isDecrease) {
+          caveats = caveats.addCaveat(
+            'erc721MaxLoss',
+            tokenAddress,
```

Comment:
I can't find the branch for these contracts, but just confirming they are different to the ERC-20 ones and are token first before recipient?

Reply:
https://github.com/MetaMask/delegation-framework/compare/main...hackathon/enforce-simulation